horizon/facts/router.py

```diff
@@ -27,11 +28,9 @@ async def create_user(
         obj_key=body.get("key"),
         authorization_header=request.headers.get("Authorization"),
     )
-    logger.info(f"Created user id: {data_entry}")
-    if await ws_client.publish(data_entry.topics, data=data_entry):
-        logger.info(f"Published user id: {data_entry}")
-    else:
-        logger.warning(f"Failed to publish user id: {data_entry}")
+    await update_subscriber.publish_and_wait(
+        data_entry, timeout=sidecar_config.LOCAL_FACTS_WAIT_TIMEOUT
+    )
```

Comment:
we have a need to let the customer choose if he want to wait for it or not, I would add an header / query param / something similar to identify if you want to call publish or publish_and_wait.

In the SDK it would look similar to 
```python
async with permit.synchronized() as c:
  await c.api.users.sync(user)
```

In go it would probably be similar to:
```golang
result := permit.Synchronized().Api.Users.Sync(user)
```

If you have a better interface that doesn't force us to add optional argument to every function it is welcomed :)

Reply:
Well, that's some different product requirements. That's a good idea and I agree that we should have some opt-in / opt-out of this feature, but that requires some additional specs.

Reply:
Ok, lets do this in the next iteration